helion/_compiler/generate_ast.py

```diff
@@ -97,6 +99,60 @@ def lift(self, expr: ast.AST, *, dce: bool = False, prefix: str = "v") -> ast.Na
             )
             return create(ast.Name, id=varname, ctx=ast.Load())
 
+    def lift_symnode(
+        self,
+        expr: ast.AST,
+        sym_expr: sympy.Expr,
+        *,
+        dce: bool = False,
+        prefix: str = "symnode",
+    ) -> ast.Name:
```

Comment:
This function hoists symnode definitions outside of the loop to make it more efficient, as suggested by @v0i0.